data-prepper-api/src/main/java/org/opensearch/dataprepper/model/failures/DlqObject.java

```diff
@@ -0,0 +1,160 @@
+        checkNotNull(failedData, "failedData cannot be null");
+
+        this.pluginId = pluginId;
+        this.pluginName = pluginName;
+        this.pipelineName = pipelineName;
+        this.version = VERSION;
+        this.failedData = failedData;
+
+        this.timestamp = StringUtils.isEmpty(timestamp) ? FORMATTER.format(Instant.now()) : timestamp;
+    }
+    
+    public String getPluginId() {
+        return pluginId;
+    }
+    
+    public String getPluginName() {
+        return pluginName;
+    }
+    
+    public String getPipelineName() {
+        return pipelineName;
+    }
+    
+    public String getVersion() {
+        return version;
+    }
+    
+    public Object getFailedData() {
+        return failedData;
+    }
+    
+    public String getTimestamp() {
+        return timestamp;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        final DlqObject that = (DlqObject) o;
+        return Objects.equals(version, that.getVersion())
+            && Objects.equals(failedData, that.getFailedData())
+            && Objects.equals(pluginId, that.pluginId)
+            && Objects.equals(pluginName, that.pluginName)
+            && Objects.equals(pipelineName, that.pipelineName)
+            && Objects.equals(timestamp, that.getTimestamp());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(pluginId, pluginName, pipelineName, version, timestamp, failedData);
+    }
+
+    @Override
+    public String toString() {
+        return "DlqObject{" +
+            "pluginId='" + pluginId + '\'' +
+            ", pluginName='" + pluginName + '\'' +
+            ", pipelineName='" + pipelineName + '\'' +
+            ", version='" + version + '\'' +
+            ", timestamp='" + timestamp + '\'' +
+            ", failedData=" + failedData +
+            '}';
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    public static class Builder {
+
+        private String pluginId;
+        private String pluginName;
+        private String pipelineName;
+        private Object failedData;
+
+        private String timestamp;
+
+        public Builder withPluginId(final String pluginId) {
+            this.pluginId = pluginId;
+            return this;
+        }
+
+        public Builder withPluginName(final String pluginName) {
+            this.pluginName = pluginName;
+            return this;
+        }
+
+        public Builder withPipelineName(final String pipelineName) {
+            this.pipelineName = pipelineName;
+            return this;
+        }
+
+        public Builder withFailedData(final Object failedData) {
+            this.failedData = failedData;
+            return this;
+        }
+
+        public Builder withTimestamp(final String timestamp) {
```

Comment:
It would be nice to let the caller specify an `Instant`. You can convert it internally.

```
public Builder withTimestamp(final Instant timestamp);
```